test_spack.py

```diff
@@ -0,0 +1,432 @@
+#!/usr/bin/python
+
+import copy
+import inspect
+import sys
+import subprocess
+import unittest
+
+all_machines = {'daint', 'tsa'}
+
+def run(command: str, cwd = '.'):
+    setup = ''
+    if command.startswith('spack'):
+        setup = f'source spack/share/spack/setup-env.sh &&'
+    
+    subprocess.run(f'{setup} cd {cwd} && {command}', check=True, shell=True)
+
+# For each spack test there should be at least one line of comment stating
+# why this spack command is tested and/or where this spack command is used.
+# Otherwise this may apply:
+# “All Your Tests are Terrible..." - Titus Winters & Hyrum Wright
+# https://www.youtube.com/watch?v=u5senBJUkPc&ab_channel=CppCon
+
+
+class AtlasTest(unittest.TestCase):
+    package_name = 'atlas'
+    depends_on = {'ecbuild', 'eckit'}
+    machines = all_machines
+
+
+class AtlasUtilityTest(unittest.TestCase):
+    package_name = 'atlas_utilities'
+    depends_on = {'atlas', 'eckit'}
+    machines = all_machines
+
+
+class ClawTest(unittest.TestCase):
+    package_name = 'claw'
+    depends_on = {}
+    machines = all_machines
+
+
+class CosmoTest(unittest.TestCase):
+    package_name = 'cosmo'
+    depends_on = {'cuda', 'serialbox', 'libgrib1', 'cosmo-grib-api-definitions', 'cosmo-eccodes-definitions', 'omni-xmod-pool', 'claw', 'zlib_ng', 'cosmo-dycore'}
+    machines = all_machines
+
+    def test_install_master_gpu(self):
+        # So our quick start tutorial works: https://c2sm.github.io/spack-c2sm/QuickStart.html
+        run('spack installcosmo cosmo@master%pgi cosmo_target=gpu +cppdycore')
+
+    def test_install_master_cpu(self):
+        # So our quick start tutorial works: https://c2sm.github.io/spack-c2sm/QuickStart.html
+        run('spack installcosmo cosmo@master%pgi cosmo_target=cpu ~cppdycore')
+
+    # def test_install_test(self):
+    #     # TODO: Decide if we want to integrate this test or not. It has been used lately here: From https://github.com/C2SM/spack-c2sm/pull/289
+    #     run('spack installcosmo --test=root cosmo@master%pgi')
```

Comment:
yes, but I dont think we need an additional test for this, rather make this the default. --test=root will run test_cosmo.py with the testsuite on top of the build, so I would add it to test_install_master_gpu|cpu